src/hpc/thread.c

```diff
@@ -103,10 +103,93 @@ void UnlockThreadControl(void)
 #define MAP_ANONYMOUS MAP_ANON
 #endif
 
-#ifndef HAVE_NATIVE_TLS
+#ifndef USE_NATIVE_TLS
+
+#ifdef USE_PTHREAD_TLS
+static int init_tls_key = 0;
+static pthread_key_t TLSKey;
+#ifdef USE_MACOS_PTHREAD_TLS_ASM
+static int tls_offset;
+
+#define OFFS 0x100
+#define END (-1)
+
+int cmpOpCode(unsigned char *code, int *with) {
+    int result = 0;
+    while (*with >= 0) {
+        if (*with == OFFS) {
+            result = *code;
+        } else {
+            if (*code != *with)
+                return -1;
+        }
+        code++;
+        with++;
+    }
+    return result;
+}
+
+void FindTLSOffset() {
+    // This is an idea borrowed from Mono. We test if the implementation
+    // of pthread_getspecific() uses the assembly code below. If that is
+    // true, we can replace calls to pthread_getspecific() with the
+    // matching inline assembly, allowing a significant performance boost.
+    // There are two possible implementations.
```

Comment:
This code also appears in `configure.ac`, and I think it would be wise to always keep the two copies in sync. Ideal would be if there was really only one copy, but pulling that off might be more of a hassle than it's worth it, so let's forget about that.

Next best is to add a comment that reminds the reader about this. Something like:  "This check exists in src/hpc/thread.c and configure.ac, and the two copies should always be kept in sync", added in both places.

Reply:
I added comments in both places.